pyvista/core/filters/uniform_grid.py

```diff
@@ -414,3 +415,291 @@ def image_threshold(
+        output = _get_output(alg)
+        self._change_fft_output_scalars(
+            output, self.point_data.active_scalars_name, output_scalars_name
+        )
+        return output
+
+    def high_pass(
+        self,
+        x_cutoff,
+        y_cutoff,
+        z_cutoff,
+        order=1,
+        output_scalars_name=None,
+        progress_bar=False,
+    ):
+        """Perform a butterworth high pass filter in the frequency domain.
+
+        This filter requires that the :class`UniformGrid` have complex point
+        scalars, usually generated after the :class:`UniformGrid` has been
+        converted to frequency domain by a :func:`UniformGridFilters.fft`
+        filter.
+
+        A :func:`UniformGridFilters.rfft` filter can be
+        used to convert the output back into the spatial
+        domain. This filter attenuates low frequency components.
+        Input and output are in doubles, with two components.
+
+        Parameters
+        ----------
+        x_cutoff : double
+            The cutoff frequency for the x axis.
+
+        y_cutoff : double
+            The cutoff frequency for the y axis.
+
+        z_cutoff : double
+            The cutoff frequency for the z axis.
+
+        order : int, optional
+            The order of the cutoff curve. Given from the equation
+            ``1 + (cutoff/freq(i, j))**(2*order)``.
+
+        output_scalars_name : str, optional
+            The name of the output scalars. By default this is the same as the
+            active scalars of the dataset.
+
+        progress_bar : bool, optional
+            Display a progress bar to indicate progress.
+
+        Returns
+        -------
+        pyvista.UniformGrid
+            UniformGrid with the applied high pass filter.
+
+        Examples
+        --------
+        See :ref:`image_fft_perlin_example` for a full example using this filter.
+
+        """
+        self._check_fft_scalars()
+        alg = _vtk.vtkImageButterworthHighPass()
+        alg.SetInputDataObject(self)
+        alg.SetCutOff(x_cutoff, y_cutoff, z_cutoff)
+        alg.SetOrder(order)
+        _update_alg(alg, progress_bar, 'Performing High Pass Filter')
+        output = _get_output(alg)
+        self._change_fft_output_scalars(
+            output, self.point_data.active_scalars_name, output_scalars_name
+        )
+        return output
+
+    def _change_fft_output_scalars(self, dataset, orig_name, out_name):
+        """Modify the name and dtype of the output scalars for a FFT filter."""
+        name = orig_name if out_name is None else out_name
+        pdata = dataset.point_data
+        if pdata.active_scalars_name != name:
+            pdata[name] = pdata.pop(pdata.active_scalars_name)
+
+        # always view the datatype of the point_data as complex128
+        dataset._association_complex_names['POINT'].add(name)
+
+    def _check_fft_scalars(self):
+        """Check for active scalars with two components.
```

Comment:
I'm a bit lost again here, because still I think we have single-component complex scalars, even according to the doctest output for `fft()`. I'm surpised the checks in this helper don't raise then.

Reply:
Fixed in 6ce17e5590254f7e030298d5bcb4c42cd0e46ca4

Reply:
Not intended behavior. This should be fixed in a separate PR as it's related to complex scalars and not the FFT filter.

Reply:
Resolving as tracked in https://github.com/pyvista/pyvista/issues/2873